File: src/main/java/com/codecool/javapeno/erp/entities/Transaction.java
package com.codecool.javapeno.erp.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.sql.Timestamp;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "transactions")
public class Transaction {
    @Id
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid")
    private UUID id;

    @NotNull
    private Timestamp timestamp;

    @NotNull
    private BigDecimal amount;

    @NotNull
    private String accountNumFrom;

    @NotNull
    private String accountNumTo;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    private User user;

}

File: src/components/Transactions/Transaction/Transaction.js
import React from 'react';

const Transaction = (props) => {
  return (
    <tr>
      <td>{props.transaction.id}</td>
      <td>{props.transaction.account_num_from}</td>
      <td>{props.transaction.account_num_to}</td>
      <td>{props.transaction.amount}</td>
      <td>{props.transaction.timestamp}</td>
      <td>{props.transaction.user_id}</td>
    </tr>
  );
};

export default Transaction;
